doc/IDE_SETUP.md

```diff
@@ -0,0 +1,170 @@
+
+```shell
+Connecting to remote nREPL server...
+Clojure 1.10.1
+```
+
+To confirm you have a working connection with REPL enter the following command in the input box below the output:
+
+```shell
+(prn 1)
+```
+
+Which should output
+
+```shell
+(prn 1)
+1
+=> nil
+```
+
+Or
+
+```shell
+prn 1
+```
+
+Which should output
+
+```shell
+prn 1
+=> 1
+=> #function[clojure.core/prn]
+```
+
+See below:
+
+![](https://notes.status.im/uploads/upload_0f39b27d0ad287db7a5f1f180f9c55b5.png)
+
+### Connecting REPL and IntelliJ to `status-react`
+
+**The important next step is telling REPL the context in which to interact with the code.**
+
+Do the following:
+
+Ensure you have 3 terminals running the following
+
+- `make run-clojure`
+- `make run-metro`
+- `make run-ios`
+
+// TODO Add the details of selecting the
+```clojure
+(shadow/watch :mobile)
+(shadow/repl :mobile)
+```
+- switch clj to cljs
+- You do need to switch it manually
+
+### Connecting Cursive / IntelliJ IDEA to REPL Problems
+
+I can connect to REPL inside the IDE, but the code doesn't seem to be able to send to REPL. But when trying to load any code into REPL I get a `Cannot load Clojure form into ClojureScript REPL`.
+
+My process:
+
+Open IDEA, not yet connected to REPL and not running a terminal with `make run-clojure`.
+
+![](https://notes.status.im/uploads/upload_a2abb27acbee62a0ca9ffdc41d2219dc.png)
+
+Attempting to connect to REPL without `make run-clojure` gives the expected error. After a terminal is running `make run-clojure`, connection works fine.
+
+Entering simple functions gives a response.
+
+![](https://notes.status.im/uploads/upload_9b2848993feb4181ad8fee96fc0dd6e9.png)
+
+However trying to do any thing REPL-ly with the code always gives a "*Cannot load Clojure form into ClojureScript REPL*" error. 
+
+![](https://notes.status.im/uploads/upload_d29b3ded54afdfddf0a2cb09bfdbd121.png)
+
+![](https://notes.status.im/uploads/upload_c07a7805e1e7c954e3d3698063410bb5.png)
+
+---
+
+### Things I've also tried
+
+I've done some Googling to see what could be going wrong.
+
+
+https://github.com/cursive-ide/cursive/issues/1285 This issue suggests switching the file to `cljc` which I did out of curiosity, it connects better but it doesn't really work.
+
+![](https://notes.status.im/uploads/upload_79bca0edc0c1841a7cbdcec50a4f90eb.png)
+
+Obviously this isn't ideal even if it worked perfectly, I presume that the file types are important and converting them back and forth is a terrible work flow.
+
+I tried converting the files to `clj`, just to see if that helped, which gave an interesting error:
+
+```bash=
+Loading src/status_im/reloader.clj... 
+Syntax error (FileNotFoundException) compiling at (src/status_im/reloader.clj:1:1).
+Could not locate status_im/ui/components/react__init.class, status_im/ui/components/react.clj or status_im/ui/components/react.cljc on classpath. Please check that namespaces with dashes use underscores in the Clojure file name.
+```
```

Comment:
Are these two sections needed since the sections above show how to connect to the repl?

Reply:
Yeah, I've removed this section